src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -209,6 +190,31 @@ namespace
     };
 }
 
+void MainWindow::new_project()
+{
+    m_project_manager.create_project();
+    on_project_change();
+}
+
+void MainWindow::open_project(const QString& filepath)
+{
+    save_state_before_project_open();
+
+    if (m_rendering_manager.is_rendering())
+    {
+        m_rendering_manager.abort_rendering();
+        m_rendering_manager.wait_until_rendering_end();
+    }
+
+    remove_render_widgets();
+
+    set_file_widgets_enabled(false, NotRendering);
+    set_project_explorer_enabled(false);
+    set_rendering_widgets_enabled(false, NotRendering);
+
+    m_project_manager.load_project(filepath.toAscii().constData());
+}
+
 void MainWindow::open_and_render_project(const QString& filepath, const QString& configuration)
```

Comment:
I'm really not enthusiastic about it. I see no value in duplicating what the project manager already offers. I'm in favor of removing this method.